File: craft/commands/referral.py
import click

from craft.client import get
from craft.output import print_item


@click.command("code")
def referral_code():
    """Get your referral code."""
    data = get("/referral")
    print_item(data)


@click.command("stats")
def referral_stats():
    """Get referral statistics and earnings."""
    data = get("/referral/stats")
    print_item(data)


@click.command("check")
@click.argument("code")
def referral_check(code):
    """Validate a referral code (no auth required)."""
    data = get(f"/referral/check/{code}", auth=False)
    print_item(data)

File: cmd/referral.go
package cmd

import (
	"fmt"

	"github.com/CraftIntertech/craft-cli/internal/client"
	"github.com/CraftIntertech/craft-cli/internal/output"
	"github.com/spf13/cobra"
)

var referralCodeCmd = &cobra.Command{
	Use:   "code",
	Short: "Show your referral code",
	Run: func(cmd *cobra.Command, args []string) {
		result := client.Get("/referral", nil, true)
		output.PrintItem(extractObj(result))
	},
}

var referralStatsCmd = &cobra.Command{
	Use:   "stats",
	Short: "Show referral statistics",
	Run: func(cmd *cobra.Command, args []string) {
		result := client.Get("/referral/stats", nil, true)
		output.PrintItem(extractObj(result))
	},
}

var referralCheckCmd = &cobra.Command{
	Use:   "check [CODE]",
	Short: "Check if a referral code is valid",
	Args:  cobra.ExactArgs(1),
	Run: func(cmd *cobra.Command, args []string) {
		code := args[0]
		result := client.Get(fmt.Sprintf("/referral/check/%s", code), nil, false)
		output.PrintItem(extractObj(result))
	},
}

func init() {
	ReferralCmd.AddCommand(referralCodeCmd)
	ReferralCmd.AddCommand(referralStatsCmd)
	ReferralCmd.AddCommand(referralCheckCmd)
}
